### ISSUE TITLE
bug: focus trapping should be disabled for sheet modals

### ISSUE THREAD

**Issue by EinfachHans** [ionic-team/ionic-framework#25147]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x
- [ ] Nightly

### Current Behavior

Ok this one is strange 😃 I experienced a bug, that i was able to reproduce very easy.

When tapping a Toast the Ion Input gets focused (and keyboard opens on mobile).

### Expected Behavior

To not happen 😃 

### Steps to Reproduce

See the linked repo. This repo was created for another issue, but can be used here as well. Steps:

1. Run App
2. Click on filled button "Bieten" -> Sheet moves to upper breakpoint and reveals the input
3. Click again on the button --> Toast gets displayed
4. Click on the Toast --> **Input gets focuses**

### Code Reproduction URL

https://github.com/EinfachHans/ionic-modal-breakpoints-example

### Ionic Info

Ionic:

   Ionic CLI                     : 6.19.0 (/Users/hans/.nvm/versions/node/v16.13.1/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 6.1.0
   @angular-devkit/build-angular : 13.2.6
   @angular-devkit/schematics    : 13.2.6
   @angular/cli                  : 13.2.6
   @ionic/angular-toolkit        : 6.1.0

Capacitor:

   Capacitor CLI      : 3.4.3
   @capacitor/android : not installed
   @capacitor/core    : 3.4.3
   @capacitor/ios     : not installed

Utility:

   cordova-res (update available: 0.15.4) : 0.15.1
   native-run                             : 1.5.0

System:

   NodeJS : v16.13.1 (/Users/hans/.nvm/versions/node/v16.13.1/bin/node)
   npm    : 8.4.0
   OS     : macOS Monterey

**Comment by liamdebeasi** [comment]
Thanks for the issue. Focus trapping is enabled with sheet modals. When you focus the toast, focus is moved back inside of the sheet.

Focus trapping should probably be disabled for sheet modals since it can overlay interactive content.
